outputs/minio/minio.go

```diff
@@ -10,26 +10,85 @@ import (
 
 	miniosdk "github.com/minio/minio-go/v7"
 
-	"github.com/falco-talon/falco-talon/internal/events"
 	minio "github.com/falco-talon/falco-talon/internal/minio/client"
+	"github.com/falco-talon/falco-talon/internal/models"
 	"github.com/falco-talon/falco-talon/internal/rules"
-	"github.com/falco-talon/falco-talon/outputs/model"
 	"github.com/falco-talon/falco-talon/utils"
 )
 
-type Config struct {
+const (
+	Name        string = "s3"
+	Category    string = "minio"
+	Description string = "Store on Minio"
+	Permissions string = ``
+	Example     string = `- action: Get logs of the pod
+  actionner: kubernetes:download
+  parameters:
+    tail_lines: 200
+  output:
+    target: minio:s3
+    parameters:
+      bucket: falco-talon
+      prefix: /files
+`
+)
+
+const (
+	defaultContentType string = "text/plain; charset=UTF-8"
+)
+
+type Parameters struct {
 	Bucket string `mapstructure:"bucket" validate:"required"`
 	Prefix string `mapstructure:"prefix" validate:""`
 }
 
-const (
-	defaultContentType string = "text/plain; charset=utf-8"
-)
+type Output struct{}
+
+func Register() *Output {
+	return new(Output)
+}
+
+func (o Output) Init() error { return minio.Init() }
 
-func Output(output *rules.Output, data *model.Data) (utils.LogLine, error) {
-	parameters := output.GetParameters()
-	var config Config
-	err := utils.DecodeParams(parameters, &config)
+func (o Output) Information() models.Information {
+	return models.Information{
+		Name:        Name,
+		FullName:    Category + ":" + Name,
+		Category:    Category,
+		Description: Description,
+		Permissions: Permissions,
+		Example:     Example,
+	}
+}
+func (o Output) Parameters() models.Parameters {
+	return Parameters{
+		Prefix: "",
+		Bucket: "",
+	}
+}
+
+func (o Output) Checks(output *rules.Output) error {
+	var parameters Parameters
+	err := utils.DecodeParams(output.GetParameters(), &parameters)
+	if err != nil {
+		return err
+	}
+
+	ctx := context.Background()
+	exist, err := minio.GetClient().BucketExists(ctx, parameters.Bucket)
+	if err != nil {
+		return err
+	}
+	if exist {
```

Comment:
the BucketExists() methods returns an error if the API call fails, if not, the value `exist` is true only if the bucket exists, if not, it contains false. This is why we need to check its content.

Reply:
Yes, it's just for readability. 

Instead of checking if exist is true and returning nil and returning the error on the bottom, I'd check if it doesn't exist and trigger the error. 

The success scenario would be at the bottom with return nil as in my suggestion.